hub/api/dataset.py

```diff
@@ -32,32 +37,36 @@ def __init__(
             mode (str): Mode in which the dataset is opened.
                 Supported modes include ("r", "w", "a") plus an optional "+" suffix.
                 Defaults to "a".
-            ds_slice (slice): The slice object restricting the view
-                of this dataset's tensors. Defaults to slice(None, None, None).
-                Used internally for iteration.
-            provider (StorageProvider, optional): The storage provider used to access
-                the data stored by this dataset.
+            ds_slice (slice): The slice object restricting the view of this dataset's tensors.
+                Defaults to slice(None, None, None). Used internally for iteration.
+            memory_cache_size (int): The size of the memory cache to be used in MB.
```

Comment:
you mention here "in MB" but the `constants.py` file has no knowledge of this fact, please update the `constants.py` file to have MB multiplication